src/EditorFeatures/Core/Implementation/SmartIndent/AbstractIndentationService.cs

```diff
@@ -51,12 +55,42 @@ private IEnumerable<IFormattingRule> GetFormattingRules(Document document, int p
                 return null;
             }
 
+            return GetBlankLineIndentation(document, lineNumber, indentStyle, cancellationToken);
+        }
+
+        /// <summary>
+        /// Determines indentation for a blank line (i.e. after hitting enter at the end of a line,
+        /// or after moving to a blank line). This indent style cannot be <see cref="FormattingOptions.IndentStyle.None"/>;
+        /// </summary>
+        public IndentationResult GetBlankLineIndentation(
+            Document document, int lineNumber, FormattingOptions.IndentStyle indentStyle, CancellationToken cancellationToken)
```

Comment:
note that this function cannot actually fail (i.e. return 'null').  It will always give a desired indentation for any line as if it was blank.

Reply:
probably worth mentioning in the doc comment

Reply:
see above.